src/prefect/task_worker.py

```diff
@@ -76,8 +77,9 @@ def __init__(
         limit: Optional[int] = 10,
     ):
         self.tasks: List[Task] = list(tasks)
+        self.task_keys = set(t.task_key for t in tasks if isinstance(t, Task))
```

Comment:
The refactoring in this file is just to make the `/status` endpoint more useful and isn't part of the fix here